packages/testing/src/child-process.ts

```diff
@@ -22,15 +26,18 @@ export async function waitOnChild(child: ChildProcess): Promise<void> {
   });
 }
 
-export async function kill(child: ChildProcess, signal: NodeJS.Signals = 'SIGINT'): Promise<void> {
+export async function kill(child: ChildProcess, signal: NodeJS.Signals = 'SIGINT', opts?: WaitOptions) {
   if (child.pid === undefined) {
     throw new TypeError('Expected child with pid');
   }
   process.kill(child.pid, signal);
   try {
-    await waitOnChild(child);
-  } catch (err: any) {
-    if (!(err.name === 'ChildProcessError' && err.signal === signal)) {
+    await waitOnChild(child, opts);
+  } catch (err) {
+    // Should error if the error is not a child process error or it is a child
+    // process and either the platform is Windows or the signal matches.
```

Comment:
Explaining why would be good here

Reply:
Hmm.. I don't remember, I copied this from another file.
@cretz added this when he worked on adding windows support.

Reply:
Likely because the `signal` field is not populated on the `exit` event in Windows